lib/hocon/impl/tokens.rb

```diff
@@ -146,14 +150,30 @@ def self.new_string(origin, value)
     new_value(ConfigString.new(origin, value))
   end
 
+  def self.new_int(origin, value, original_text)
+    new_value(ConfigInt.new(origin, value, original_text))
```

Comment:
For consistency with the `new_long` method below and the Java code, I think it would be best to call `ConfigNumber.new_number` here and in the below `new_float` method.